src/service/history-impl.js

```diff
@@ -329,19 +352,15 @@ export class HistoryBindingNatural_ {
     history.pushState = this.historyPushState_.bind(this);
     history.replaceState = this.historyReplaceState_.bind(this);
 
-    const eventPass = new Pass(this.win, this.onHistoryEvent_.bind(this), 50);
     this.popstateHandler_ = e => {
+      if (this.ignoreUpcomingPopstate_) {
+        return;
+      }
       dev().fine(TAG_, 'popstate event: ' + this.win.history.length + ', ' +
           JSON.stringify(e.state));
-      eventPass.schedule();
-    };
-    this.hashchangeHandler_ = () => {
```

Comment:
Dropping hashchange handler as it is not needed now that we handle our own hash-navigations when users click on anchors. And after investigating browsers, they all fire `popstate` on navigating backward or forward in history.